wallet_anchor.go

```diff
@@ -54,8 +54,13 @@ func (l *LndRpcWalletAnchor) FundPsbt(ctx context.Context, packet *psbt.Packet,
 
 	pkt, changeIndex, leasedUtxos, err := l.lnd.WalletKit.FundPsbt(
 		ctx, &walletrpc.FundPsbtRequest{
-			Template: &walletrpc.FundPsbtRequest_Psbt{
-				Psbt: psbtBuf.Bytes(),
+			Template: &walletrpc.FundPsbtRequest_CoinSelect{
+				CoinSelect: &walletrpc.PsbtCoinSelect{
+					Psbt: psbtBuf.Bytes(),
+					ChangeOutput: &walletrpc.PsbtCoinSelect_Add{
```

Comment:
There was some loss of context due to the creation of a new issue. But this isn't enough. What we want to get rid of is this code (or simplify it):

https://github.com/lightninglabs/taproot-assets/blob/52f174888ec4c92e619a3c996772700875afeb56/tapfreighter/wallet.go#L1445-L1486

and

https://github.com/lightninglabs/taproot-assets/blob/3305aef4b2b079685f218b786200f83545e900bc/tapgarden/caretaker.go#L657-L671

See original comment here: https://github.com/lightninglabs/taproot-assets/issues/118#issue-1372070303

In the first case (`wallet.go`), we can add the anchor PSBT inputs before funding and can then skip the `adjustFundedPsbt` step (because the new `FundPsbt` call doesn't do BIP-69 ordering anymore).

In the second case (`caretaker.go`) we are now sure that the change output index is always at index 1, so we don't need that extra logic anymore.

Reply:
> In the second case (caretaker.go) we are now sure that the change output index is always at index 1, so we don't need that extra logic anymore.

If I ask `FundPsbt` to always place change on index 1 I'll cause many itests to fail on [`MintAssetsConfirmBatch`](https://github.com/lightninglabs/taproot-assets/blob/main/itest/assets_test.go#L99)

```
2024-03-26 19:37:30.119 [ERR] RPCS: [/mintrpc.Mint/FinalizeBatch]: unable to finalize batch: unable to advance state machine: unable to fund psbt: unable to fund psbt: rpc error: code = Unknown desc = change output index out of range: 1
```

do we want to create the new index, or check if it exists and place there?

Reply:
You need to add a change output first (derive address) and add it, so there is something there at index 1.